examples/directory-bucket/main.tf

```diff
@@ -0,0 +1,121 @@
+locals {
+  region = "eu-west-1"
+}
+
+provider "aws" {
+  region = local.region
+
+  # Make it faster by skipping something
+  skip_metadata_api_check     = true
+  skip_region_validation      = true
+  skip_credentials_validation = true
+}
+
+data "aws_caller_identity" "current" {}
+
+module "simple" {
+  source = "../../"
+
+  is_directory_bucket = true
+  bucket              = random_pet.this.id
+  availability_zone   = "${local.region}b"
+}
+
+module "complete" {
+  source = "../../"
+
+  is_directory_bucket = true
+  bucket              = "${random_pet.this.id}-complete"
+  availability_zone   = "${local.region}b"
```

Comment:
As I mentioned earlier. `b` or other AZs may not be present in some AWS accounts. I hit this issue in the past myself. Please update the example to use data-source to find first-available AZ in the account.

Reply:
Oh I see, my apologies, I misunderstood initially. I added a data source to find available AZs but I found that even though an AZ is available to me, it might not be an option for directory buckets. In the example, if I set it to index `0`, it attempts to use `euw1-az2` which is available for my account but not available for directory buckets so I used index `1` here. https://docs.aws.amazon.com/AmazonS3/latest/userguide/s3-express-Endpoints.html

I guess this may not work for everyone so I'm happy to modify this.

Reply:
Thank you! 👍